packages/react/src/oidc/core/routes/OidcRoutes.tsx

```diff
@@ -22,6 +23,7 @@ type OidcRoutesProps = {
   configurationName:string;
   redirect_uri: string;
   silent_redirect_uri?: string;
+  withCustomHistory?: () => CustomHistory;
```

Comment:
Awesome thank you for your PullRequest,

Do you have a sample of your implementation of that fonction in NextJS ?

Reply:
Yes,

Here is an excerpt, it's very simple:


```typescript
const MyApp: React.FC<AppProps> = ({ Component, pageProps: { session, ...pageProps }, router }) => {
  const [loading, setLoading] = useState(router.asPath.includes('access_token'));

  const store = useStore(pageProps.initialReduxState);
  let searchRedirectPage: PageUrl;

  const withCustomHistory: () => CustomHistory = () => {
    return {
      replaceState: (url?: string | null, stateHistory?: WindowHistoryState): void => {
        router.replace({
          pathname: url,
        });
      },
    };
  };
  // Code omitted...

  return !loading ? (
    <>
      <Head>
        <meta
          name="viewport"
          content="width=device-width, height=device-height,  initial-scale=1.0, user-scalable=no;user-scalable=0;"
        />
      </Head>
      <OidcProvider configuration={OIDC_CONFIGURATION} withCustomHistory={withCustomHistory}>
        <Provider store={store}>
          <RouterScrollProvider>{layout}</RouterScrollProvider>
        </Provider>
      </OidcProvider>
    </>
  ) : null;
};

```

Reply:
this would be awesome to add more docmentation and a demo after that PullRequest.

Thank you very much !